packages/circuits/src/email_auth_template.circom

```diff
@@ -182,6 +182,191 @@ template EmailAuth(n, k, max_header_bytes, max_subject_bytes, recipient_enabled)
             recipient_email_addr[i] <== shifted_email_addr[i] * has_email_recipient;
         }
     
+        signal recipient_email_addr_ints[num_email_addr_ints] <== Bytes2Ints(email_max_bytes)(recipient_email_addr);
+        signal recipient_email_addr_commit_raw;
+        recipient_email_addr_commit_raw <== EmailAddrCommit(num_email_addr_ints)(cm_rand, recipient_email_addr_ints);
+        recipient_email_addr_commit <== has_email_recipient * recipient_email_addr_commit_raw;
+    }
+}
+
+// Verify email from user (sender) and extract subject, timestmap, recipient email (commitment), etc.
+// * n - the number of bits in each chunk of the RSA public key (modulust)
+// * k - the number of chunks in the RSA public key (n * k > 2048)
+// * max_header_bytes - max number of bytes in the email header
+// * max_body_bytes - max number of bytes in the email body
+// * max_command_bytes - max number of bytes in the command
+// * recipient_enabled - whether the email address commitment of the recipient = email address in the subject is exposed
+// * is_qp_encoded - whether the email body is qp encoded
+template EmailAuthWithBodyParsing(n, k, max_header_bytes, max_body_bytes, max_command_bytes, recipient_enabled, is_qp_encoded) {
```

Comment:
> is_qp_encoded

Can we fix the value of `is_qp_encoded` when operating the service?
Or do we need to deploy two circuits and choose which circuit is used for each email?

Reply:
The second option, because the flag will be turned on for email with big commands (which have soft-line breaks) and should be turned off for small ones as the proving time improves.

Reply:
Do you think a malicious relayer can choose an inappropriate circuit, which can be an attack vector?

Reply:
If the reason to turn off the flag is only for efficiency, I think we can always use a circuit with soft-line breaks.